for-developers/walkthrough/dev-setup/making-your-first-changes.md

```diff
@@ -47,4 +47,1930 @@ Now you should see some custom content in the post that you added to the feed. Y
+```go
+RuleErrorReactEntryAlreadyExists                       RuleError = "RuleErrorReactEntryAlreadyExists"
+RuleErrorCannotReactNonexistentPost                    RuleError = "RuleErrorCannotReactNonexistentPost"
+RuleErrorCannotRemoveReactionWithoutAnExistingReaction RuleError = "RuleErrorCannotRemoveReactionWithoutAnExistingReaction"
+```
+
+#### `blockchain.go`
+In `blockchain.go`, add a `CreateReactTxn` method after the `CreateLikeTxn` implementation.
+
+```go
+func (bc *Blockchain) CreateReactTxn(
+	userPublicKey []byte, postHash BlockHash, isRemove bool, emojiReaction rune,
+	minFeeRateNanosPerKB uint64, mempool *DeSoMempool, additionalOutputs []*DeSoOutput) (
+	_txn *MsgDeSoTxn, _totalInput uint64, _changeAmount uint64, _fees uint64,
+	_err error) {
+
+	// A React transaction doesn't need any inputs or outputs (except additionalOutputs provided).
+	txn := &MsgDeSoTxn{
+		PublicKey: userPublicKey,
+		TxnMeta: &ReactMetadata{
+			PostHash:      &postHash,
+			EmojiReaction: emojiReaction,
+			IsRemove:      isRemove,
+		},
+		TxOutputs: additionalOutputs,
+		// We wait to compute the signature until we've added all the
+		// inputs and change.
+	}
+
+	totalInput, spendAmount, changeAmount, fees, err :=
+		bc.AddInputsAndChangeToTransaction(txn, minFeeRateNanosPerKB, mempool)
+	if err != nil {
+		return nil, 0, 0, 0, errors.Wrapf(
+			err, "CreateReactTxn: Problem adding inputs: ")
+	}
+
+	// Sanity-check that the spendAmount is zero.
+	if err = amountEqualsAdditionalOutputs(spendAmount, additionalOutputs); err != nil {
+		return nil, 0, 0, 0, fmt.Errorf("CreateReactTxn: %v", err)
+	}
+
+	return txn, totalInput, changeAmount, fees, nil
+}
+```
+
+### Creating custom reaction files 
+#### `block_view_reaction.go`
+`block_view_reaction.go` is inspired from `block_view_like.go`.
+It defines setters and getters to the state generated by a `React` transaction to the UTXO view,
+and implements custom `_connectReact` and `disconnectReact` transactions.
+
+```go
+package lib
+
+import (
+	"fmt"
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"reflect"
+)
+
+func (bav *UtxoView) _getReactionEntryForReactionKey(reactionKey *ReactionKey) *ReactionEntry {
+	// If an entry exists in the in-memory map, return the value of that mapping.
+	mapValue, existsMapValue := bav.ReactionKeyToReactionEntry[*reactionKey]
+	if existsMapValue {
+		return mapValue
+	}
+
+	// If we get here it means no value exists in our in-memory map. In this case,
+	// defer to the db. If a mapping exists in the db, return it. If not, return
+	// nil. Either way, save the value to the in-memory view mapping got later.
+	reactionExists := false
+	if bav.Postgres != nil {
+		reactionExists = bav.Postgres.GetReaction(reactionKey.ReactorPubKey[:], &reactionKey.ReactedPostHash) != nil
+	} else {
+		reactionExists = DbGetReactorPubKeyToPostHashMapping(bav.Handle, reactionKey.ReactorPubKey[:], reactionKey.ReactedPostHash) != nil
+	}
```

Comment:
We recently introduced the concept of a `DbAdapter` to handle switching between badger and postgres when querying for data